src/server/pfs/s3/server.go

```diff
@@ -22,74 +54,136 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
```

Comment:
It might make sense to have this be separate handlers. I think that'd be slightly less code and make it a little more organized, over time each of these blocks is probably going to grow so we might as well do it now. I think the built in router for the `http` package should have everything you need but if not feel free to bring in something like: https://github.com/gorilla/mux

Reply:
http's builtin router only does literal prefix matching, which would help a bit but not much. I brought in mux, which has made things a lot cleaner.